sdk/src/main/java/io/horizen/account/state/ExecutionRevertedException.java

```diff
@@ -1,23 +1,25 @@
 package io.horizen.account.state;
 
+import org.web3j.utils.Numeric;
+
 /**
  * Message processing failed, also revert-and-keep-gas-left.
  */
 public class ExecutionRevertedException extends ExecutionFailedException {
-    public final byte[] revertReason;
+    public final byte[] returnData;
 
-    public ExecutionRevertedException(byte[] revertReason) {
-        super("execution reverted");
-        this.revertReason = revertReason;
+    public ExecutionRevertedException(byte[] returnData) {
+        super(String.format("execution reverted with return data \"%s\"", Numeric.toHexString(returnData)));
+        this.returnData = returnData;
     }
 
     public ExecutionRevertedException(String message) {
         super(message);
-        this.revertReason = new byte[0];
+        this.returnData = new byte[0];
     }
 
     public ExecutionRevertedException(String message, Throwable cause) {
         super(message, cause);
-        this.revertReason = new byte[0];
+        this.returnData = new byte[0];
```

Comment:
Maybe a constant can be defined, instead of creating a new object every time (eg `static final byte[] EMPTY_RETURN_DATA = new byte[0];`)

Reply:
done